devices/giex.js

```diff
@@ -1,135 +1,202 @@
+                .withDescription('Number of cycle irrigation times, set to 0 for single cycle'),
+            exposes.numeric(keys.giexWaterValve.irrigationStartTime, ea.STATE)
+                .withDescription('Last irrigation start time'),
+            exposes.numeric(keys.giexWaterValve.irrigationEndTime, ea.STATE)
+                .withDescription('Last irrigation end time'),
+            exposes.numeric(keys.giexWaterValve.lastIrrigationDuration, ea.STATE)
+                .withDescription('Last irrigation duration'),
+            exposes.numeric(keys.giexWaterValve.waterConsumed, ea.STATE)
+                .withUnit('L')
+                .withDescription('Last irrigation water consumption'),
+        ],
+    },
+};
+
 module.exports = [
+    // _TZE200_a7sghmms uses seconds, timezone is local
     {
+        ...exportTemplates.giexWaterValve,
+        model: 'QT06_1',
         fingerprint: [
-            {modelID: 'TS0601', manufacturerName: '_TZE200_sh1btabb'},
             {modelID: 'TS0601', manufacturerName: '_TZE200_a7sghmms'},
         ],
-        model: 'QT06',
-        vendor: 'GiEX',
-        description: 'Water irrigation valve',
-        onEvent: tuya.onEventSetLocalTime,
-        fromZigbee: [fzLocal.giex_water_valve],
-        toZigbee: [tzLocal.giex_water_valve],
         exposes: [
-            e.battery(),
-            exposes.binary('state', ea.STATE_SET, 'ON', 'OFF').withDescription('State'),
-            exposes.enum('mode', ea.STATE_SET, ['duration', 'capacity']).withDescription('Irrigation mode'),
-            exposes.numeric('irrigation_target', exposes.access.STATE_SET).withValueMin(0).withValueMax(3600).withUnit('seconds or litres')
+            ...exportTemplates.giexWaterValve.exposes,
+            exposes.numeric(keys.giexWaterValve.irrigationTarget, ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(SECONDS_IN_12_HOURS)
+                .withUnit('seconds or litres')
                 .withDescription('Irrigation target, duration in seconds or capacity in litres (depending on mode)'),
-            exposes.numeric('cycle_irrigation_num_times', exposes.access.STATE_SET).withValueMin(0).withValueMax(100)
-                .withDescription('Number of cycle irrigation times, set to 0 for single cycle'),
-            exposes.numeric('cycle_irrigation_interval', exposes.access.STATE_SET).withValueMin(0).withValueMax(3600).withUnit('sec')
+            exposes.numeric(keys.giexWaterValve.cycleIrrigationInterval, ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(SECONDS_IN_12_HOURS)
+                .withUnit('sec')
+                .withDescription('Cycle irrigation interval'),
+        ],
+    },
+    // _TZE200_sh1btabb uses minutes, timezone is GMT+8
+    {
+        ...exportTemplates.giexWaterValve,
+        model: 'QT06_2',
+        fingerprint: [
+            {modelID: 'TS0601', manufacturerName: '_TZE200_sh1btabb'},
+        ],
+        timezone: '+08:00',
```

Comment:
Please don't add extra props on the definition level. I suggest to use the following in the converter:

```
 const time = model.model == 'QT06_2' ? '+08:00' : null
```

Reply:
Hmmm... I didn't want to add this model logic directly into the converter, just in case it keeps growing. I'll look for some other solution without polluting the model object, like a config object or something. 👍